tests/api-watch-diff/test_module.ml

```diff
@@ -0,0 +1,141 @@
+open Api_watch_diff
+open Test_helpers
+
+let%expect_test "Modules with both value and submodule changes" =
+  let reference =
+    compile_interface
+      {|
+    type ('a, 'b) result = Ok of 'a | Error of 'b
+    val f : int -> string
+    module M : sig
+      val g : int -> string
+    end
+  |}
+  in
+  let current =
+    compile_interface
+      {|
+    type ('a, 'b) result = Ok of 'a | Error of 'b
+    val f : int -> (string, string) result
+    module M : sig
+      val g : int -> (string, string) result
+    end
+  |}
+  in
+  let result = diff_interface ~module_name:"Main" ~reference ~current in
+  Format.printf "%a" pp_diff_list result;
+  [%expect
+    {|
+    Some Module Main: [ Value (f, Modified);
+    Module M: [ Value (g, Modified);]]|}]
+
+let%expect_test "Modules with multiple value and submodule changes" =
+  let reference =
+    compile_interface
+      {|
+    type ('a, 'b) result = Ok of 'a | Error of 'b
+    val a : string -> int 
+    val f : int -> string
+    module M : sig
+      val b : int list -> int
+      val g : int -> string
+      val z : string
+    end
+  |}
+  in
+  let current =
+    compile_interface
+      {|
+    type ('a, 'b) result = Ok of 'a | Error of 'b
+    val a : string -> float
+    val f : int -> (string, string) result
+    module M : sig
+      val b : float list -> float
+      val g : int -> (string, string) result
+      val z : string
+    end
+    module N : sig val x: int end
+  |}
+  in
+  let result = diff_interface ~module_name:"Main" ~reference ~current in
+  Format.printf "%a" pp_diff_list result;
+  [%expect
+    {|
+    Some Module Main: [ Value (a, Modified); Value (f, Modified);
+    Module M: [ Value (b, Modified); Value (g, Modified);]
+    Module N: Unsupported changes]|}]
```

Comment:
Same here, I think we should still be able to detect added and removed modules at least. It's ok to report unsupported changes when two versions of a module are not both `Mty_signatures` but those simple changes we should be able to detect.